gutils/gutils.c

```diff
@@ -28,70 +28,22 @@
 #include "basics.h"
 #include "gutils.h"
 
-#include <stdio.h>
-#include <math.h>
-#include <stdlib.h>
-#include <string.h>
-#include <ustring.h>
-#include <utype.h>
-#include <unistd.h>
-#include <locale.h>
-#if !defined(__MINGW32__)
-# include <pwd.h>
-#endif
-#include <stdarg.h>
-#include <time.h>
-#include <gdraw.h>		/* For image defn */
-
-#ifdef __CygWin
- #include <sys/types.h>
- #include <sys/stat.h>
- #include <unistd.h>
-#endif
-
+#include <ffglib.h>
 
 const char *GetAuthor(void) {
-#if defined(__MINGW32__)
-    static char author[200] = { '\0' };
-    if ( author[0] == '\0' ){
-	char* name = getenv("USER");
-	if(!name) return NULL;
-	strncpy(author, name, sizeof(author));
-	author[sizeof(author)-1] = '\0';
-    }
-    return author;
-#else
-    struct passwd *pwd;
-    static char author[200] = { '\0' };
-    const char *ret = NULL, *username;
+    static char author[200] = {0};
 
-    if ( author[0]!='\0' )
-return( author );
-
-/* Can all be commented out if no pwd routines */
-    pwd = getpwuid(getuid());
-    username = getenv("USER");
-    if (getenv("SOURCE_DATE_EPOCH") && username) {
-	strncpy(author, username, sizeof(author));
-	author[sizeof(author)-1] = '\0';
-	ret = author;
-    } else if ( pwd!=NULL && pwd->pw_gecos!=NULL && *pwd->pw_gecos!='\0' ) {
-	strncpy(author,pwd->pw_gecos,sizeof(author));
-	author[sizeof(author)-1] = '\0';
-	ret = author;
-    } else if ( pwd!=NULL && pwd->pw_name!=NULL && *pwd->pw_name!='\0' ) {
-	strncpy(author,pwd->pw_name,sizeof(author));
-	author[sizeof(author)-1] = '\0';
-	ret = author;
-    } else if ( username !=NULL ) {
-	strncpy(author, username, sizeof(author));
-	author[sizeof(author)-1] = '\0';
-	ret = author;
+    if (author[0]) {
+        return author;
+    } else if (getenv("SOURCE_DATE_EPOCH")) {
+        const char *username = getenv("USER");
+        if (username) {
+            snprintf(author, sizeof(author), "%s", username);
+            return author;
+        }
```

Comment:
I take it it doesn't matter that this will be enabled and have priority under `__MINGW32_`?

Reply:
Yeah that's fine, from what I understand, the SOURCE_DATE_EPOCH+USER env variables were to always have higher priority for reproducibility (and there wasn't a mandate that this only applies to linux)

Reply:
lgtm